test-harness/src/test/groovy/com/netflix/conductor/test/integration/NestedForkJoinSubworkflowRestartSpec.groovy

```diff
@@ -0,0 +1,468 @@
+package com.netflix.conductor.test.integration
+
+import com.netflix.conductor.common.metadata.tasks.Task
+import com.netflix.conductor.common.metadata.tasks.TaskDef
+import com.netflix.conductor.common.run.Workflow
+import com.netflix.conductor.core.execution.tasks.Fork
+import com.netflix.conductor.core.execution.tasks.SubWorkflow
+import com.netflix.conductor.core.execution.tasks.WorkflowSystemTask
+import com.netflix.conductor.dao.QueueDAO
+import com.netflix.conductor.test.base.AbstractSpecification
+import org.springframework.beans.factory.annotation.Autowired
+import spock.lang.Shared
+
+import static com.netflix.conductor.common.metadata.tasks.TaskType.TASK_TYPE_JOIN
+import static com.netflix.conductor.common.metadata.tasks.TaskType.TASK_TYPE_SUB_WORKFLOW
+import static com.netflix.conductor.test.util.WorkflowTestUtil.verifyPolledAndAcknowledgedTask
+
+class NestedForkJoinSubworkflowRestartSpec extends AbstractSpecification {
+
+    @Shared
+    def FORK_JOIN_NESTED_SUB_WF = 'nested_fork_join_swf'
```

Comment:
I think we need a case where there are nested fork joins with the subworkflow contained in the inner fork.

Reply:
Good catch. I will also add some cases where the SUB_WORKFLOW task is optional.